### DISCUSSION TITLE
How to use EventCallbacks/provide examples and Popup Customization

### DISCUSSION THREAD

**Discussion by sanongpost** [gismofx/toast_ui.blazor_calendar#28] [General]
@gismofx 

Pls kindly advise, possibly by adding to current sample for customization such as callback event like below, only to see how calendar item can be intercept.

callback;
======
-OnClickCalendarEventOrTask
-OnChangeCalendarEventOrTask
-OnCreateCalendarEventOrTask
-OnDeleteCalendarEventOrTask

popup
====
-can this be resize or customize  ? for example, calendar is in <div> or MudPaper, the popup size stay the same.

regards

sanong

i will be retrieving data / save to SQL server, and do some validation etc when user do something on screen.

**Comment by gismofx** [comment]
@sanongpost Have a look at the tui calendar docs. There is a lot to customize that you can control. https://nhn.github.io/tui.calendar/latest/Options

If you want to use your own pop-up editors you can disable the default ones with these properties in this class:
https://github.com/gismofx/toast_ui.blazor_calendar/blob/5cf78683218cafbecd3cf1f46f41645e7fe153d2/toast_ui.blazor_calendar/Models/TUICalendarOptions.cs#L66 
(I haven't done this yet, but someone else might have)

Events/Callbacks, you can browse the code and code comments from intellisense. That said, I should and will add some additional samples into the sample project which use all the available events so you can get an idea how they should be used.

**Comment by sanongpost** [comment]
Thanks  ***@***.***> @gismofx/toast_ui.blazor_calendar

 

We appreciate existing sample with added usage of the component :

-event callback

-sizing and positioning which I have tried with Mudblazor ( eg. MudPaper) and found that dialog size don’t go along.

 

I am sure developer can extend from your guide easily.

 

Thank you keep supporting.

 

Regards

 

Sanong

**Comment by gismofx** [comment]
@sanongpost 

Examples added to sample project with how to use the events emitted from the calendar:
Here:
https://github.com/gismofx/toast_ui.blazor_calendar/blob/475e30f09ca01ff3210a4f439bc512221369e690/toast_ui.blazor_calendar.TestProject/toast_ui.blazor_calendar.TestProject/Pages/Index.razor#L39-L49

and here:
https://github.com/gismofx/toast_ui.blazor_calendar/blob/475e30f09ca01ff3210a4f439bc512221369e690/toast_ui.blazor_calendar.TestProject/toast_ui.blazor_calendar.TestProject/ViewModels/CalendarViewModel.cs#L171-L195

If you want to use your own pop-up editor you can, but you'll need to play with the settings and logic on your own.. Maybe read thru Tui there may be some customization you can do with their pop-up.
